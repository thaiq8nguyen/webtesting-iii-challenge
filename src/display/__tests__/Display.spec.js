import React from "react";
import { cleanup, render } from "@testing-library/react";
import Display from "../Display";

afterEach(cleanup);

/* SNAPSHOT TESTING */
test("creating controller state snapshot using snapshot testing", () => {
  const { asFragment } = render(<Display />);
  expect(asFragment()).toMatchSnapshot();
});

test("display lock status and close state when locked=false and closed=false", () => {
  const { getByText } = render(<Display locked={false} closed={false} />);

  const lockStatusSign = getByText("Unlocked");
  const closeStatusSign = getByText("Open");

  expect(lockStatusSign.classList.contains("green-led")).toBeTruthy();
  expect(closeStatusSign.classList.contains("green-led")).toBeTruthy();
});

test("display lock status and close state when locked=false and closed=true", () => {
  const { getByText } = render(<Display locked={false} closed={true} />);

  const lockStatusSign = getByText("Unlocked");
  const closeStatusSign = getByText("Closed");

  expect(lockStatusSign.classList.contains("green-led")).toBeTruthy();
  expect(closeStatusSign.classList.contains("red-led")).toBeTruthy();
  //console.log(closeStatusSign.classList);
});
