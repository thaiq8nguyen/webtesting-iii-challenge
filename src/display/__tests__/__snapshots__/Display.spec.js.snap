// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creating controller state snapshot using snapshot testing 1`] = `
<DocumentFragment>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
    >
      Unlocked
    </div>
    <div
      class="led green-led"
    >
      Open
    </div>
  </div>
</DocumentFragment>
`;
